docs/rules/custom-rules.md

```diff
@@ -131,61 +157,99 @@ Notice we don't need to include `property` in this approach.
 ```yaml
 rules:
   assert/path-item-operation-required:
-    subject: PathItem
+    subject:
+      type: PathItem
+    assertions:
+      required:
+        - get
     message: Every path item must have a GET operation.
-    required:
-      - get
 ```
 
 The following example asserts that Tags have both name and description defined.
 
 ```yaml
 rules:
   assert/tag-name-and-desc-defined:
-    subject: Tag
-    property:
-      - name
-      - description
+    subject:
+      type: Tag
+      property:
+        - name
+        - description
+    assertions:
+      defined: true
     message: Every tag must have a name and description.
-    defined: true
 ```
 
 Another way to compose that rule is to require the subject keys:
 
 ```yaml
 rules:
   assert/tag-name-and-desc-required:
-    subject: Tag
-    message: Every tag must have a name and description.
-    required:
-      - name
-      - description
+    subject:
+      type: Tag
+    assertions:
+      required:
+        - name
+        - description
 ```
 
-### `context` example
+### `where` example
 
 The following example asserts that PUT responses with HTTP status 200 or 201 cannot return an `application/pdf`content type.
-Without the `context`, the assertion would evaluate every MediaTypesMap including:
-- Responses with all codes, including codes other than 200 or 201
+Without the `where`, the assertion would evaluate every `MediaTypesMap` property including:
+- Responses with all codes, including codes other than 200 or 201.
 - Responses for all HTTP methods, including DELETE, GET, POST, and more.
-To restrict the evaluation, use the `context` feature to limit what will be evaluated.
+To restrict the evaluation, use the `where` feature to limit what is evaluated.
 
 ```yaml
 assert/no-pdf-in-ok-response:
-  context:
-  - type: Operation
-    matchParentKeys: [put]
-  - type: Response
-    matchParentKeys: ['201', '200']
-  subject: MediaTypesMap
-  disallowed: ['application/pdf']
+  where:
+  - whereSubject:
+      type: Operation
+      filterByParentKeys: 
+        - put
+    whereAssertions:
+      defined: true
+  - whereSubject:
+      type: Response
+      filterByParentKeys: 
+        - '201'
+        - '200'
+    whereAssertions:
+      defined: true
+  subject:
+    type: MediaTypesMap
+  assertions:
+    disallowed:
+      - 'application/pdf'
+```
+
+Where enables complex assertions based on sibling values.
+The following example asserts that the `limit` parameter must have a schema with `type: integer`.
+
+```yaml
+assert/limit-is-integer:
+  subject:
+    type: Schema
+    property: type
+  assertions:
+    enum:
+      - integer
+  where:
+    - whereSubject:
+        type: Parameter
+        property: name
+      whereAssertions:
+        enum:
+          - limit
```

Comment:
what do you think about using a shorter version of lists to make examples more compact?

```suggestion
assert/limit-is-integer:
  subject:
    type: Schema
    property: type
  assertions:
    enum: [integer]
  where:
    - whereSubject:
        type: Parameter
        property: name
      whereAssertions:
        enum: [limit]
```

Reply:
I don't like this approach because it has inconsistency and has some unwritten and uncommon rule.

Reply:
Okay, agreed.